[2022-12-08 20:12:20,020][    INFO] {'criterion': {'kwargs': {'use_weight': False}, 'type': 'CELoss'},
 'dataset': {'batch_size': 2,
             'ignore_label': 255,
             'mean': [123.675, 116.28, 103.53],
             'n_sup': 195,
             'noise_std': 0.1,
             'std': [58.395, 57.12, 57.375],
             'train': {'GaussianBlur': False,
                       'crop': {'size': [128, 128], 'type': 'rand'},
                       'data_list': '../../../../data/splits/pascal/1464/labeled.txt',
                       'data_root': '../../../../data/VOC2012',
                       'flip': True,
                       'rand_resize': [0.5, 2.0]},
             'type': 'pascal_semi',
             'val': {'crop': {'size': [128, 128], 'type': 'center'},
                     'data_list': '../../../../data/splits/pascal/val.txt',
                     'data_root': '../../../../data/VOC2012'},
             'workers': 1},
 'exp_path': '',
 'net': {'decoder': {'kwargs': {'dilations': [12, 24, 36], 'inner_planes': 256},
                     'type': 'u2pl.models.decoder.dec_deeplabv3_plus'},
         'ema_decay': 0.99,
         'encoder': {'kwargs': {'fpn': True,
                                'multi_grid': True,
                                'replace_stride_with_dilation': [False,
                                                                 True,
                                                                 True],
                                'zero_init_residual': True},
                     'type': 'u2pl.models.resnet.resnet101'},
         'num_classes': 9,
         'sync_bn': True},
 'save_path': 'checkpoints101_rgb',
 'saver': {'pretrain': '', 'snapshot_dir': 'checkpoints101_rgb'},
 'trainer': {'contrastive': {'current_class_negative_threshold': 1,
                             'current_class_threshold': 0.3,
                             'high_rank': 20,
                             'low_entropy_threshold': 20,
                             'low_rank': 3,
                             'negative_high_entropy': True,
                             'num_negatives': 101,
                             'num_queries': 256,
                             'temperature': 0.5,
                             'unsupervised_entropy_ignore': 80},
             'epochs': 80,
             'eval_on': True,
             'lr_scheduler': {'kwargs': {'power': 0.9}, 'mode': 'poly'},
             'optimizer': {'kwargs': {'lr': 0.001,
                                      'momentum': 0.9,
                                      'weight_decay': 0.0001},
                           'type': 'SGD'},
             'unsupervised': {'TTA': False,
                              'apply_aug': 'cutmix',
                              'drop_percent': 80}}}
set random seed to 2
[Info] Load ImageNet pretrain from '/home/jidian/sdb1/jinlei/U2PL/prtain/resnet101.pth' 
missing_keys:  [] 
unexpected_keys:  ['fc.weight', 'fc.bias']
[2022-12-08 20:12:23,242][    INFO] # samples: 202
[2022-12-08 20:12:23,247][    INFO] # samples: 1418
[2022-12-08 20:12:23,250][    INFO] # samples: 180
[2022-12-08 20:12:23,250][    INFO] Get loader Done...
[Info] Load ImageNet pretrain from '/home/jidian/sdb1/jinlei/U2PL/prtain/resnet101.pth' 
missing_keys:  [] 
unexpected_keys:  ['fc.weight', 'fc.bias']
[2022-12-08 20:12:25,665][    INFO] The kwargs for lr scheduler: 0.9
Traceback (most recent call last):
  File "../../../../train_semi.py", line 477, in <module>
    main()
  File "../../../../train_semi.py", line 144, in main
    train( model, model_teacher, optimizer,lr_scheduler,sup_loss_fn, train_loader_sup,train_loader_unsup,epoch,tb_logger,logger,memobank,queue_ptrlis,queue_size,)
  File "../../../../train_semi.py", line 373, in train
    loss.backward()
  File "/home/jidian/anaconda3/envs/u2pl/lib/python3.6/site-packages/torch/tensor.py", line 221, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph)
  File "/home/jidian/anaconda3/envs/u2pl/lib/python3.6/site-packages/torch/autograd/__init__.py", line 132, in backward
    allow_unreachable=True)  # allow_unreachable flag
RuntimeError: CUDA out of memory. Tried to allocate 4.25 GiB (GPU 0; 11.91 GiB total capacity; 1.33 GiB already allocated; 4.25 GiB free; 2.10 GiB reserved in total by PyTorch)
Traceback (most recent call last):
  File "/home/jidian/anaconda3/envs/u2pl/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/jidian/anaconda3/envs/u2pl/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/jidian/anaconda3/envs/u2pl/lib/python3.6/site-packages/torch/distributed/launch.py", line 260, in <module>
    main()
  File "/home/jidian/anaconda3/envs/u2pl/lib/python3.6/site-packages/torch/distributed/launch.py", line 256, in main
    cmd=cmd)
subprocess.CalledProcessError: Command '['/home/jidian/anaconda3/envs/u2pl/bin/python', '-u', '../../../../train_semi.py', '--local_rank=0', '--config=config1.yaml', '--seed', '2', '--port', '1255']' returned non-zero exit status 1.
